typed/remove.go

```diff
@@ -118,7 +126,17 @@ func (w *removingWalker) doMap(t *schema.Map) ValidationErrors {
 		// but ignore them when we are removing (i.e. !w.shouldExtract)
 		if w.toRemove.Has(path) {
 			if w.shouldExtract {
-				newMap[k] = val.Unstructured()
+				valIsAtomic, err := isAtomic(val, w.schema, fieldType)
```

Comment:
Let me know if we can just ignore this err and don't have to do the whole errors collection rigamarole. I'm thinking we can because [`resolveSchema()`](https://github.com/kubernetes-sigs/structured-merge-diff/pull/188/files#diff-9bfd5d4edd67f21a522024f4f5725b30dffdceada9af3fee0310ab699755401cR44) does the `schema.Resolve(typeRef)` call that has the potential to error out, and we just ignore it in that case.